doc/releases/changelog-dev.md

```diff
@@ -17,6 +17,13 @@
 * Alias for Identity (`I`) is now accessible from `qml.ops`.
   [(#7200)](https://github.com/PennyLaneAI/pennylane/pull/7200)
 
+<h3>Labs: a place for unified and rapid prototyping of research software 🧪</h3>
+
+* A `parity_matrix` function is now available in `pennylane.labs.phase_polynomials`.
+  It allows to compute the efficient intermediate representation of CNOT circuits in terms of its parity matrix
+  and is important for CNOT routing algorithms and other quantum compilation routines.
```

Comment:
Maybe focusing a bit more on the key objects, and split the two statements about what it does, and why one would like to do it :)
```suggestion
  It allows to compute the parity matrix of a CNOT circuit, which is an efficient intermediate representation.
  It is important for CNOT routing algorithms and other quantum compilation routines.
```